### ISSUE TITLE
Deferred deep link is not working on Android (SDK version 4.29.5)

### ISSUE THREAD

**Issue by polcats** [adjust/react_native_sdk#192]
The deferred deep link is not working. I am currently using SDK version 4.29.5.

Here's how I am testing it on Android:

I have set up deferred deep links in an app and released it the internal test track of Google Play.

I wanted to confirm if its getting detected, so I put an alert in the listener:
```ts
    adjustConfig.setShouldLaunchDeeplink(true);
    adjustConfig.setDeferredDeeplinkCallbackListener((uri) => {
      Alert.alert('setDeferredDeeplinkCallbackListener', JSON.stringify(uri));
    });
```

I created a deep link in the Adjust dashboard via the Deep Link generator. 

This link opens the Play Store page of the app when it gets opened in an Android device (Samsung Galaxy A73 5G).

I installed the app, then opened it, but the deep link is not getting detected? Or somehow, the listener is not working?

A sample of the generated deep link I am using is:
```
https://app.adjust.com/hx2mojm?deep_link=my-app%3A%2F%2Fitem%3Fslug%3Dsample-slug%26cat%3Dsample_category
```

Is there any other way to test?

Other details: 
- The app already has a deep link handling set up (foregrounded and backgrounded cases), so I know that its working.
- Also, if the app is already installed, and the deep link gets pressed, it's getting detected/parsed correctly by the app.
- The app version in the internal test track is the one that's visible to me in the store because I am set as one of the internal testers. I have confirmed that it is the one that I configured with the `setDeferredDeeplinkCallbackListener`.

TLDR; the deferred deep link listener does not seem to be working, and I am looking for alternative ways to debug it.

**Comment by polcats** [comment]
Update: This issue is resolved. I had to "forget" the device in the testing console, before re-doing the whole test scenario.
